### PR TITLE
api-client-java:0.6.0

### PR DESCRIPTION
**Pull request by mikemccarty-vertex**
## Summary

- Updated to latest `platform.vertexvis.com/spec` API spec
- Incremented library version to `0.6.0`
- Updated api generator to `openapi-generator-cli:v5.4.0`
- Added serdes support for metadata value types

### CODE REVIEW

File: src/test/java/com/vertexvis/model/AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeTest {
+
```
Comment (mikemccarty-vertex) [inline src/test/java/com/vertexvis/model/AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeTest.java:13]: Tests support for serialization and deserialization of metadata property values.

File: src/main/java/com/vertexvis/model/serialization/AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeAdapter.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    @Override
+    public void write(JsonWriter out,
+                      AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullType value)
+            throws IOException {
+        out.jsonValue(gsonSupplier.get().toJson(value.getValue()));
+    }
+
+    @Override
+    public AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullType read(
+            JsonReader in) throws IOException {
```
Comment (mikemccarty-vertex) [inline src/main/java/com/vertexvis/model/serialization/AnyOfMetadataLongTypeMetadataFloatTypeMetadataDateTypeMetadataStringTypeMetadataNullTypeAdapter.java:39]: Deserialization of the metadata type.  Made more complicated by the `anyOf` qualifier in the schema.  The `type` field always wins in the case of ambiguity.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
```
Comment (danschultz) [inline gradle/wrapper/gradle-wrapper.properties:3]: 💯

File: scripts/generate.sh [hunk 4]
```diff
@@ -27,8 +31,8 @@ main() {
   for f in "${models[@]}"; do mv "src/main/java/com/vertexvis/model/$f" . || true; done
 
   rm -rf api gradle src
-  
-  docker run --rm -v "${PWD}:/local" openapitools/openapi-generator-cli:v5.2.1 generate \
+
+  docker run --rm -v "${PWD}:/local" openapitools/openapi-generator-cli:v5.4.0 generate \
```
Comment (danschultz) [inline scripts/generate.sh:35]: 💯

**Pull request merged**